File: lib/models/workout.dart
import 'package:fitr/models/exercise.dart';
import 'package:fitr/models/session.dart';

class Workout {
  int id;
  String name;
  int userId;
  bool isActive;
  List<Session> sessions;
  List<Exercise> exercises;

  Workout(
      {this.id,
      this.name,
      this.userId,
      this.isActive,
      this.sessions,
      this.exercises});

  factory Workout.fromJson(Map<String, dynamic> json) {
    var listOfExercises = json['exercises'] as List;
    var exercises = listOfExercises.map((ex) => Exercise.fromJson(ex)).toList();

    return Workout(
        id: json['id'],
        name: json['name'],
        userId: json['userId'],
        isActive: json['isActive'],
        sessions: json['sessions'],
        exercises: exercises);
  }
}

File: api/Fitr/Models/Workout.cs
using System.Collections.Generic;

namespace Fitr.Models
{
    public class Workout
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string UserId { get; set; }
        
        public virtual User User { get; set; }

        public bool IsActive { get; set; }

        public virtual ICollection<Session> Sessions { get; set; }

        public virtual ICollection<WorkoutHasExercise> Exercises { get; set; }
    }
}